chia/cmds/wallet_funcs.py

```diff
@@ -132,415 +131,454 @@ async def get_unit_name_for_wallet_id(
     return name
 
 
-async def get_transaction(args: dict, wallet_client: WalletRpcClient, fingerprint: int) -> None:
-    transaction_id = bytes32.from_hexstr(args["tx_id"])
-    config = load_config(DEFAULT_ROOT_PATH, "config.yaml", SERVICE_NAME)
-    address_prefix = selected_network_address_prefix(config)
-    # The wallet id parameter is required by the client but unused by the RPC.
-    this_is_unused = 37
-    tx: TransactionRecord = await wallet_client.get_transaction(this_is_unused, transaction_id=transaction_id)
-
-    try:
-        wallet_type = await get_wallet_type(wallet_id=tx.wallet_id, wallet_client=wallet_client)
-        mojo_per_unit = get_mojo_per_unit(wallet_type=wallet_type)
-        name = await get_unit_name_for_wallet_id(
-            config=config,
-            wallet_type=wallet_type,
-            wallet_id=tx.wallet_id,
-            wallet_client=wallet_client,
+async def get_transaction(wallet_rpc_port: Optional[int], fingerprint: Optional[int], tx_id: str, verbose: int) -> None:
+    async with get_wallet_client(wallet_rpc_port, fingerprint) as (wallet_client, fingerprint, config):
+        if wallet_client is None:
+            return
```

Comment:
instead of checking this condition everywhere you use this context manager, what would you think of throwing an exception instead? I think it would simplify using the context manager, and avoid the risk of forgetting this check

Reply:
we have the weird none thing so we can cleanly output a pretty version of the traceback

Reply:
> we have the weird none thing so we can cleanly output a pretty version of the traceback

I would happily get rid of it but i think product would be mad.